src/backend/base/langflow/api/v1/jobs.py

```diff
@@ -0,0 +1,175 @@
+            run_at=None,
+            name=request.name,
+            kwargs={
+                "flow": flow,
+                "input_request": request.input_request,
+                "stream": False,
+                "api_key_user": user,
+            },
+        )
+    except Exception as exc:
+        raise HTTPException(status_code=500, detail=str(exc)) from exc
+
+
+@router.get("/{job_id}", response_model=JobRead)
+async def get_job(
+    job_id: str,
+    user: CurrentActiveUser,
+) -> JobRead:
+    """Retrieve details for a specific job by its unique identifier.
+
+    This endpoint fetches the job data from the database, verifying ownership by the
+    currently authenticated user. If the job does not exist or the user does not own the job,
+    a 404 is raised.
+
+    Args:
+        job_id (str): The unique identifier of the job to retrieve.
+        user (CurrentActiveUser): The currently authenticated user.
+
+    Returns:
+        JobRead: The job details, including status, name, and timestamps.
+
+    Raises:
+        HTTPException:
+            - 404 if the job does not exist or is not owned by the current user.
+    """
+    task_service: JobsService = get_jobs_service()
+    task = await task_service.get_job(job_id, user.id)
+    if not task:
+        raise HTTPException(status_code=404, detail="Job not found")
+    logger.info(f"Job: {task}")
+    async with session_scope() as session:
+        if not user.id:
+            msg = "User not found"
+            raise HTTPException(status_code=404, detail=msg)
+        stmt = select(Job).where(Job.id == job_id, Job.user_id == user.id)
+        return (await session.exec(stmt)).first()
+
+
+@router.get("/", response_model=list[JobRead])
+async def get_jobs(
+    user: CurrentActiveUser,
+    task_service: Annotated[JobsService, Depends(get_jobs_service)],
+    pending: bool | None = None,
+    status: JobStatus | None = None,
+) -> list[JobRead]:
+    """Retrieve a list of jobs associated with the current user.
+
+    This endpoint allows filtering jobs based on their 'pending' state (an APSJob.job_state attribute)
+    and/or their status in the database. If 'pending' is not provided, it will simply return all
+    available jobs for the user, optionally filtered by a specified status.
+
+    Args:
+        user (CurrentActiveUser): The currently authenticated user.
+        task_service (JobsService): The service used to perform job-related operations.
+        pending (Optional[bool]): When provided, filters jobs by whether they are pending.
+        status (Optional[JobStatus]): When provided, filters jobs by a specific status
+            (e.g., PENDING, COMPLETED, FAILED, etc.).
+
+    Returns:
+        list[JobRead]: A list of job records matching the given criteria.
+    """
+    if pending is not None:
+        jobs = await task_service.get_jobs(user_id=user.id, pending=pending, status=status)
+        ids = [job.id for job in jobs]
```

Comment:
Why do we do a second query ?
Can't the result of `task_service.get_jobs` be returned directly ?